src/App.tsx

```diff
@@ -2,24 +1,64 @@
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import Pagination from './components/Pagination/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const total = 42;
+const items = getNumbers(1, total).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const handleChangePerPages = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
+
+  const calculatePageRange = (
+    currentPageArg: number,
+    perPageArg: number,
+    totalArg: number,
+  ) => {
+    const startElement = (currentPageArg - 1) * perPageArg + 1;
+    const endElement = Math.min(currentPageArg * perPageArg, totalArg);
+
+    return { startElement, endElement };
+  };
+
+  // Использование:
+  const { startElement, endElement } = calculatePageRange(
+    currentPage,
+    perPage,
+    total,
+  );
+
+  const showItems = items.slice(startElement - 1, endElement).map(item => (
```

Comment:
When slicing the `items` array, you're correctly adjusting the `startElement` by subtracting 1 to accommodate the zero-based index of arrays in JavaScript. Good attention to detail.